pkg/ccn-lite/patches/0001-fix-compiler-warnings.patch

```diff
@@ -0,0 +1,202 @@
+From 35a3a6484448acc7a48c5b084daab356d79a0100 Mon Sep 17 00:00:00 2001
+From: smlng <s@mlng.net>
+Date: Sun, 15 Jan 2017 17:57:34 +0100
+Subject: [PATCH] fix compiler warnings
+
+---
+ src/ccnl-core-fwd.c | 16 ++++++++--------
+ src/ccnl-core.c     | 32 +++++++++++++++++---------------
+ 2 files changed, 25 insertions(+), 23 deletions(-)
+
+diff --git a/src/ccnl-core-fwd.c b/src/ccnl-core-fwd.c
+index 59689cf..616161d 100644
+--- a/src/ccnl-core-fwd.c
++++ b/src/ccnl-core-fwd.c
+@@ -78,10 +78,10 @@ ccnl_fwd_handleContent(struct ccnl_relay_s *relay, struct ccnl_face_s *from,
+     // Find the original prefix for the intermediate result and use that prefix to cache the content.
+     if (ccnl_nfnprefix_isIntermediate((*pkt)->pfx)) {
+         if (ccnl_nfn_RX_intermediate(relay, from, pkt)) {
+-            // This was an intermediate result from the compute server. 
++            // This was an intermediate result from the compute server.
+             // It was cached, and shouldn't be forwarded.
+             DEBUGMSG_CFWD(VERBOSE, "received intermediate result from compute server \n");
+-            return 0;   
++            return 0;
+         }
+     }
+ #endif
+@@ -123,7 +123,7 @@ ccnl_fwd_handleContent(struct ccnl_relay_s *relay, struct ccnl_face_s *from,
+         if (relay->max_cache_entries != 0) { // it's set to -1 or a limit
+             DEBUGMSG_CFWD(DEBUG, "  adding content to cache\n");
+             ccnl_content_add2cache(relay, c);
+-    	    DEBUGMSG_CFWD(INFO, "data after creating packet %.*s\n", c->pkt->contlen, c->pkt->content);
++            DEBUGMSG_CFWD(INFO, "data after creating packet %.*s\n", c->pkt->contlen, c->pkt->content);
+         } else {
+             DEBUGMSG_CFWD(DEBUG, "  content not added to cache\n");
+             free_content(c);
+@@ -201,7 +201,7 @@ ccnl_fwd_handleInterest(struct ccnl_relay_s *relay, struct ccnl_face_s *from,
+     int nonce = 0;
+     if (pkt != NULL && (*pkt) != NULL && (*pkt)->s.ndntlv.nonce != NULL) {
+         if ((*pkt)->s.ndntlv.nonce->datalen == 4) {
+-            nonce = *((int*)(*pkt)->s.ndntlv.nonce->data);
++            memcpy(&nonce, (*pkt)->s.ndntlv.nonce->data, 4);
```

Comment:
the original code causes a `strict-alignment` error by the compiler. `memcpy` is one possible fix, another would be to change the struct behind this <- hence, I found using `memcpy` less intrusive. 

In general, it seems rather opportunistic to assume an `int` has size 4 on any platform, or not? So additionally, it might be better to define `int32_t nounce = 0` as I suggested above.